src/generate-CST-derivings.sh

```diff
@@ -0,0 +1,125 @@
+        CST_DERIVERS='[@@deriving yojson]'
+        ;;
+
+    visitors)
+        PRELUDE='(** Visitors for concrete syntax trees.
+
+   This module contains all the visitors that can be generated using
+   {{: https://gitlab.inria.fr/fpottier/visitors }  the PPX visitors library },
+   namely:
+   {ul {li iter} {li map} {li reduce} {li mapreduce} {li iter2} {li map2} {li reduce2}}
+
+   The generated type for visitors on the whole concrete syntax tree can be
+   huge, which makes them unreadable and the files containing them ridiculously
+   big. They are therefore not generated and not shown here. We refer the user to
+   {{: http://gallium.inria.fr/~fpottier/visitors/manual.pdf } the documentation of visitors }. *)
+
+(* The following disables odoc for the rest of the file. *)
+(**/**)'
+        INTRO_DERIVERS='
+[@@deriving
+  visitors { variety = "iter";      name = "located_iter";      polymorphic = true },
+  visitors { variety = "map";       name = "located_map";       polymorphic = true },
+  visitors { variety = "reduce";    name = "located_reduce";    polymorphic = true },
+  visitors { variety = "mapreduce"; name = "located_mapreduce"; polymorphic = true },
+  visitors { variety = "iter2";     name = "located_iter2";     polymorphic = true },
+  visitors { variety = "map2";      name = "located_map2";      polymorphic = true },
+  visitors { variety = "reduce2";   name = "located_reduce2";   polymorphic = true }
+]'
+        CST_DERIVERS='
+[@@deriving
+  visitors { variety = "iter";       ancestors = ["located_iter"];      nude = true },
+  visitors { variety = "map";        ancestors = ["located_map"];       nude = true },
+  visitors { variety = "reduce";     ancestors = ["located_reduce"];    nude = true },
+  visitors { variety = "mapreduce";  ancestors = ["located_mapreduce"]; nude = true },
+  visitors { variety = "iter2";      ancestors = ["located_iter2"];     nude = true },
+  visitors { variety = "map2";       ancestors = ["located_map2"];      nude = true },
+  visitors { variety = "reduce2";    ancestors = ["located_reduce2"];   nude = true }
+]'
+        ;;
+
+    *)
+        printf 'I do not know what to do with kind %s.\n' "$KIND"
+        exit 2
+esac
+
+## The following sed recognizes anything of the form:
+##
+##     type foo = ...
+##     type foo = bar = ...
+##     and foo = ...
+##     and foo = bar = ...
+##
+## and replaces them with:
+##
+##     type foo = $1.foo = ...
+##     type foo = $1.foo = ...
+##     and foo = $1.foo = ...
+##     and foo = $1.foo = ...
+##
+## where $1 is the first argument, except when what follows the equal sign
+## starts with a lowercase, because this would then be a syntax equality and we
+## must not add an other one. This function is useful to copy a whole type while
+## keeping it equal with the original module's type.
+
+add_type_equalities () {
+    sed "s/\(type\|and\)\(.* \)\([a-zA-Z_'][a-zA-Z_']*\) *=\([^=]*=\)\? *\([^ a-z]\|$\)/\1\2\3 = \2CST.\3 = \5/g"
```

Comment:
If I understand correctly, you are specifying the language for type identifiers in the first occurrence (right after the `type`) while the type identifier between the two equals is less constrained. Any reason for that? 

(I am not asking for a change, just curious.)

Reply:
No really good reason for that, no, it just took me a while to write this regular expression and I might have changed my strategy in the middle. I think I like the OCaml version (of #128, which I wrote afterwards) better.